code/modules/atmospherics/machinery/components/machines/crystallizer.dm

```diff
@@ -0,0 +1,334 @@
+#define MIN_PROGRESS_AMOUNT 				3
+#define MIN_DEVIATION_RATE 					0.90
+#define MAX_DEVIATION_RATE 					1.1
+#define HIGH_EFFICIENCY_CONDUCTIVITY 		0.95
+
+/obj/machinery/atmospherics/components/binary/crystallizer
+	icon = 'icons/obj/atmospherics/components/machines.dmi'
+	icon_state = "crystallizer-off"
+	name = "crystallizer"
+	desc = "Used to crystallize or solidify gases."
+	layer = ABOVE_MOB_LAYER
+	density = TRUE
+	max_integrity = 300
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, RAD = 100, FIRE = 80, ACID = 30)
+	circuit = /obj/item/circuitboard/machine/crystallizer
+	pipe_flags = PIPING_ONE_PER_TURF
+
+	///Save the icon state for the machine to be used in update_icon()
+	var/icon_state_off = "crystallizer-off"
+	var/icon_state_on = "crystallizer-on"
+	var/icon_state_open = "crystallizer-open"
```

Comment:
You could simply have `base_icon_state = "crystallizer"` and then do `"[base_icon_state-on]" for example.